airbyte-integrations/connectors/source-singlestore/src/test/java/io/airbyte/integrations/source/singlestore/SinglestoreSourceTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.singlestore;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.cdk.db.jdbc.JdbcUtils;
+import io.airbyte.commons.json.Jsons;
+import org.junit.jupiter.api.Test;
+
+public class SinglestoreSourceTest {
+
+  private static final String EXPECTED_JDBC_ESCAPED_URL = "jdbc:singlestore://localhost:1111/db%2Ffoo?";
```

Comment:
This is the first time, I see that the database name is escaped...
Probably, we need to add it to other connectors as well.